### PR TITLE
feat(cli): Add verbosity, return `Hugr` from `run`.

### PR DESCRIPTION
**Pull request by doug-q**

### CODE REVIEW

**Comment by aborgna-q** [comment]
> Is adding the dependency for this overkill?

Seems to be a wrapper over [`log`](https://crates.io/crates/log) with a helper to add the clap parameter.

It's behind the binary dependency, so it shouldn't affect other things.
And it's part of the `clap-rs` project, so it should be fine.

File: hugr/src/cli.rs [hunk 1]
```diff
@@ -38,17 +43,23 @@ pub const VALID_PRINT: &str = "HUGR valid!";
...
         if self.mermaid {
             println!("{}", hugr.mermaid_string());
         }
 
         if !self.no_validate {
             hugr.update_validate(registry)?;
-
-            println!("{}", VALID_PRINT);
+            if self.verbosity(Level::Info) {
+                eprintln!("{}", VALID_PRINT);
```
Comment (ss2165) [inline hugr/src/cli.rs:55]: why send this to stderr? Is it more appropriate than stdout?

File: hugr/tests/cli.rs [hunk 2]
```diff
@@ -91,3 +100,21 @@ fn test_bad_json(mut cmd: Command) {
...
+        .failure()
+        .stderr(contains("Error parsing input").not());
+}
+
+#[rstest]
+fn test_valid_silent(test_hugr_string: String, mut cmd: Command) {
+    cmd.arg("-");
+    cmd.write_stdin(test_hugr_string);
+
+    cmd.assert().success().stderr(contains(VALID_PRINT).not());
```
Comment (ss2165) [inline hugr/tests/cli.rs:119]: not sure I like the default behaviour being silent. I was emulating e.g. mypy which prints a success message (we could also use colours for the messages 👀 )

Comment (ss2165) [reply hugr/tests/cli.rs:119]: yeah happy with default at Info

**Review by ss2165 — APPROVED** [review]

**Pull request merged**